### PR TITLE
marketplace initial commit

### PR DESCRIPTION
**Pull request by PiranavanShanmugavadivelu**
Completed Tasks
1. Initiated tailwind CSS
2. Used Flowbite for UI
3. Developed Home page UI for desktop and mobile
4. Created components for UI

Remaining Tasks
1. Routing
2. Setup a UI component library with the vue support.(flowbite-vue is not suitable for the future)
3. Connect with API endpoints
4. Other marketplace screens UI

### CODE REVIEW

File: frontend/marketplace/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+import AuctionCard from "@/components/AuctionCard.vue";
+import Footer from "@/components/Footer.vue";
+</script>
+
+<template>
+  <NavBar/>
+
+  <div class="p-5 md:px-[10%]">
+    <!--    Search Auction-->
+    <div class="relative md:hidden z-0 w-full mb-7 mt-3">
```
Comment (ba1ciu) [inline frontend/marketplace/src/App.vue:13]: Thinking out loud. Shall we make components out of the sections we have here in `App.vue` (Search bar, Auctions, hero banner, etc.), so basically this file looks like this?
```html
<component1/>
<component2/>
<component3/>
...
```
@jschill ?

File: frontend/marketplace/.eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/* eslint-env node */
+require('@rushstack/eslint-patch/modern-module-resolution')
+
+module.exports = {
+  root: true,
+  'extends': [
+    'plugin:vue/vue3-essential',
+    'eslint:recommended',
+    '@vue/eslint-config-typescript',
+    '@vue/eslint-config-prettier/skip-formatting'
```
Comment (jschill) [inline frontend/marketplace/.eslintrc.cjs:10]: Any special reason why you have ditched the standardjs formatting? I noticed its been removed in frontend/proof-of-existence as well. Not a big deal (even though i think its good to be consistent with empower-platform), but curious if it caused any problems?

File: frontend/marketplace/src/App.vue (hunk 1 above)
Comment (jschill) [reply frontend/marketplace/src/App.vue:13]: @ba1ciu Sounds like a great suggestion. This file is quite template-heavy atm.

File: frontend/marketplace/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { fileURLToPath, URL } from 'node:url'
+
+import { defineConfig } from 'vite'
+import vue from '@vitejs/plugin-vue'
+import vueJsx from '@vitejs/plugin-vue-jsx'
```
Comment (jschill) [inline frontend/marketplace/vite.config.ts:5]: Are we planning to use jsx?

**Comment by jschill** [comment]
I think this is fine, maybe break down the components into smaller ones as @ba1ciu suggested and clean up everything jsx-related unless you're gonna use it. 

Im very curious about your impressions of tailwind & flowbite, if that is something we should bring to the platform.

File: frontend/marketplace/src/App.vue (hunk 1 above)
Comment (PiranavanShanmugavadivelu) [reply frontend/marketplace/src/App.vue:13]: Sure, I will break these as separate components.

File: frontend/marketplace/vite.config.ts (hunk 3 above)
Comment (PiranavanShanmugavadivelu) [reply frontend/marketplace/vite.config.ts:5]: No, I just added JSX support when creating the new Vue project, I will remove these

File: frontend/marketplace/.eslintrc.cjs (hunk 2 above)
Comment (PiranavanShanmugavadivelu) [reply frontend/marketplace/.eslintrc.cjs:10]: I used default eslint support from Vue when creating a new Vue project, I couldn't get your point can you explain little bit

File: frontend/marketplace/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "pinia": "^2.0.32",
+    "vue": "^3.2.47",
+    "vue-router": "^4.1.6"
+  },
+  "devDependencies": {
+    "@rushstack/eslint-patch": "^1.2.0",
+    "@types/jsdom": "^21.1.0",
+    "@types/node": "^18.14.2",
+    "@vitejs/plugin-vue": "^4.0.0",
+    "@vitejs/plugin-vue-jsx": "^3.0.0",
```
Comment (gjermundgaraba) [inline frontend/marketplace/package.json:26]:
```suggestion
```

**Review by ba1ciu — APPROVED** [review]

**Review by gjermundgaraba — APPROVED** [review]

**Pull request merged**